go-packages/meep-model/model.go

```diff
@@ -421,76 +421,203 @@ func (m *Model) UpdateNetChar(nc *dataModel.EventNetworkCharacteristicsUpdate) (
 	return err
 }
 
-func (m *Model) isValidPoaNodeType(nodeType string) bool {
-
-	switch nodeType {
-	case NodeTypePoa, NodeTypePoa4G, NodeTypePoa5G, NodeTypePoaWifi:
-		return true
-	}
-	return false
-}
-
 // AddScenarioNode - Add scenario node
 func (m *Model) AddScenarioNode(node *dataModel.ScenarioNode) (err error) {
 	m.lock.Lock()
 	defer m.lock.Unlock()
 
 	if node == nil {
-		err = errors.New("node == nil")
-		return
+		return errors.New("node == nil")
 	}
 
-	// Find parent
+	// Find & validate parent
 	parentNode := m.nodeMap.FindByName(node.Parent)
 	if parentNode == nil {
-		err = errors.New("Parent element " + node.Parent + " not found in scenario " + m.name)
-		return
+		return errors.New("Parent element " + node.Parent + " not found in scenario " + m.name)
+	}
+	if !validateParentType(node.Type_, parentNode.nodeType) {
+		return errors.New("Invalid parent type: " + parentNode.nodeType + " for node type: " + node.Type_)
 	}
 
 	// Add element based on type
-	if node.Type_ == NodeTypeUE {
+	if isPhyLoc(node.Type_) {
```

Comment:
The switch statement requires several copies of the Node Type. Since it is used in multiple locations and more error prone if we add/remove/modify types I decided to go with a function instead.

Reply:
I will check if the function length can be reduced by adding other functions.